django/db/models/expressions.py

```diff
@@ -534,7 +534,7 @@ def resolve_expression(self, query=None, allow_joins=True, reuse=None, summarize
             c.source_expressions[pos] = arg.resolve_expression(query, allow_joins, reuse, summarize, for_save)
         return c
 
-    def as_sql(self, compiler, connection, function=None, template=None):
+    def as_sql(self, compiler, connection, function=None, template=None, arg_joiner=None, **extra_context):
```

Comment:
That's actually a very good example. Thankfully, because I was struggling to come up with one even though I was fairly sure it was needed. I'll include it as an example.